### DISCUSSION TITLE
Encrypt Direct Messages

### DISCUSSION THREAD

**Discussion by jamieschuts** [mastodon/mastodon#21458] [General]
As we know, DMs can be viewed by server admins. End users are not made aware of this, especially new users. While we'd like to think server admins will never do this, unfortunately it will happen in isolated cases. There is also no mecahnism to alert users that someone has viewed or accessed their direct messages.
Propose that DMs be envelope encryption with a combination of Server key + user key. Where users generate their own key as they join the server. This key is only known to the user.

**Comment by jamieschuts** [comment]
I'll leave this up for a couple of weeks then create an issue from it. I assume that is the correct process to get this in developer backlog? please correct me if I am wrong. thanks.

**Reply by mohe2015** [reply]
https://github.com/mastodon/mastodon/issues/19565 is probably related
